components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/template/APITemplateBuilderImpl.java

```diff
@@ -79,6 +92,39 @@ public String getConfigStringForTemplate(Environment environment) throws APITemp
             configcontext = new EnvironmentConfigContext(configcontext, environment);
             configcontext = new TemplateUtilContext(configcontext);
 
+            if (api.getWsdlUrl() != null && !api.getWsdlUrl().isEmpty()) {
+                RegistryService registryService = ServiceReferenceHolder.getInstance().getRegistryService();
+                String tenantDomain = MultitenantUtils.getTenantDomain(
+                        APIUtil.replaceEmailDomainBack(api.getId().getProviderName()));
+                int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                        .getTenantId(tenantDomain);
+                String resourcePath = APIConstants.API_LOCATION + RegistryConstants.PATH_SEPARATOR +
+                        api.getId().getProviderName() + RegistryConstants.PATH_SEPARATOR + api.getId().getApiName()
+                        + RegistryConstants.PATH_SEPARATOR + api.getId().getVersion() + RegistryConstants.PATH_SEPARATOR
+                        + SOAPToRESTConstants.SEQUENCE_GEN.SOAP_TO_REST_IN_RESOURCE;
+                Resource regResource;
+
+                UserRegistry registry = registryService.getGovernanceSystemRegistry(tenantId);
+                if (registry.resourceExists(resourcePath)) {
```

Comment:
what would happen if the resource is not found in the registry? shouldn't there be a exception or at-least an error log? Please check that case and add the 'else' condition if needed